client/memory.c

```diff
@@ -93,8 +96,8 @@ int xdag_mem_init(size_t size)
 	}
 
 	size_t wrote = snprintf(tfile_node->tmpfile, PATH_MAX,"%s%s", g_tmpfile_path, TMPFILE_TEMPLATE);
-	if (wrote >= PATH_MAX){
-		xdag_fatal("Error: Temporary file path exceed the max length that is %d characters", PATH_MAX);
+	if (wrote < 0){
```

Comment:
size_t is unsigned, please cast to ssize_t

Reply:
for snprintf, the error return is less than 0, I don't think  PATH_MAX check is necessary, 
size_t should be int
```
The number of characters that would have been written if n had been sufficiently large, not counting the terminating null character.
If an encoding error occurs, a negative number is returned.
Notice that only when this returned value is non-negative and less than n, the string has been completely written.
```

Reply:
right, change it to int, 
if there was PATH_MAX check, it means there was a purpose for it.
You are adding the name of the file to the path, the path could also be already truncated here -> xdag_mem_tempfile_path but no one is checking it, if it was truncated there the PATH_MAX catch here catch it (it's just an example of why PATH_MAX check is needed).

Reply:
```
Upon successful return, these functions return the number of characters printed (excluding the null byte used to end output to strings).
The functions snprintf() and vsnprintf() do not write more than size bytes (including the terminating null byte ('\0')). If the output was truncated due to this limit then the return value is the number of characters (excluding the terminating null byte) which would have been written to the final string if enough space had been available. Thus, a return value of size or more means that the output was truncated. (See also below under NOTES.)

If an output error is encountered, a negative value is returned.
```
another version of man of snprintf, u r right.